src/Symfony/Component/RateLimiter/CompoundLimiter.php

```diff
@@ -31,6 +33,11 @@ public function __construct(array $limiters)
         $this->limiters = $limiters;
     }
 
+    public function reserve(int $tokens = 1, ?float $maxTime = null): Reservation
+    {
+        // TODO: Implement reserve() method.
+    }
```

Comment:
I didn't really know how a compound limiter should act on reserving tokens. Should it reserve a token for each limiter? That can lead to reserving tokens at all different timeslots, which timeslot should this limiter then return?

Reply:
At first, I would indeed throw an exception. That way, if we find a good way later on, we will be able to support it "easily" without any BC breaks.